components/sync15/src/sync_multiple.rs

```diff
@@ -192,6 +221,19 @@ impl<'info, 'res, 'pgs, 'mcs> SyncMultipleDriver<'info, 'res, 'pgs, 'mcs> {
         // store failing.
         self.result.service_status = ServiceStatus::Ok;
 
+        if let Some(command_processor) = self.command_processor {
+            log::info!("Synchronizing clients engine");
+            let engine = clients::Engine {
+                command_processor,
+                interruptee: self.interruptee,
+                storage_client: &client_info.client,
+                global_state: &global_state,
+                root_sync_key: &self.root_sync_key,
+            };
+            // Note that a clients engine sync failure is fatal.
+            engine.sync()?;
```

Comment:
Hmm, I think we'll still want telemetry if we fail here, and it should be easier to add than adding telemetry above (which is before a "sync" so it's unclear where to even put it...).

Reply:
I'll add this next!